src/openrct2/world/Sprite.cpp

```diff
@@ -533,10 +532,9 @@ void sprite_set_coordinates(const CoordsXYZ& spritePos, EntityBase* sprite)
 {
     auto screenCoords = translate_3d_to_2d_with_z(get_current_rotation(), spritePos);
 
-    sprite->sprite_left = screenCoords.x - sprite->sprite_width;
-    sprite->sprite_right = screenCoords.x + sprite->sprite_width;
-    sprite->sprite_top = screenCoords.y - sprite->sprite_height_negative;
-    sprite->sprite_bottom = screenCoords.y + sprite->sprite_height_positive;
+    sprite->SpriteRect = ScreenRect(
+        screenCoords.x - sprite->sprite_width, screenCoords.y - sprite->sprite_height_negative,
+        screenCoords.x + sprite->sprite_width, screenCoords.y + sprite->sprite_height_positive);
```

Comment:
```suggestion
        screenCoords - ScreenCordsXY{ sprite->sprite_width, sprite->sprite_height_negative },
        screenCoords + ScreenCordsXY{ sprite->sprite_width, sprite->sprite_height_positive });
```
Footprint is slightly smaller, and I think that there is a `ScreenRect` ctor that takes a pair of `ScreenCoordsXY`

Reply:
Hmm, i don't really like that constructor. It's not normal for a rectangle constructor to be defined as two points normally it would be a point and a pair of width height. I would think that a two screen coord constructor would be xy widthheight